### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Change issue templates to use labels instead of title prefixes, and add more content to the feature request template.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+      label: Background and Motivation
+      description: What is the problem we are solving and in what context did you encounter it?
+    validations:
+      required: true
+  - type: textarea
+    id: proposal
+    attributes:
+      label: Proposed Feature
+      description:
+        Please provide a sketch of the feature you are proposing. Be as specific as you can: the more specific the proposal, the easier the process will be. Including screenshots of some of the existing problems can also help a lot here.
```
Comment (MikaelMayer) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:31]: I'm pretty sure it's not possible add screenshots in the template version of the issue. Have you tried?

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
     validations:
       required: true
+  - type: textarea
+    id: alternatives
+    attributes:
+      label: Alternative Designs
+      description: |
+        Were there other options you considered?
+        How does this compare to features in other language ecosystems?
```
Comment (MikaelMayer) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:40]: Could you add something like this:

What is the closest workaround for this feature?
How could you live without this feature until now?

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
I like it. It will make feature proposals more comprehensive.

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml (hunk 1 above)
Comment (MikaelMayer) [reply .github/ISSUE_TEMPLATE/FEATURE_REQUEST.yml:31]: You might want to add "if you want to add screenshot, use the free-form issue form"

**Review by MikaelMayer — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: Bug Report
 description: File a bug report
-title: "[Bug]: "
-labels: []
+title: ""
```
Comment (fabiomadge) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:3]: The empty title seems to be invalid, but you can just remove it, as it's optional.

**Review by fabiomadge — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 3 above)
Comment (keyboardDrummer) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: What do you mean by it seems to be invalid?

Comment (fabiomadge) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: And calling it out now: I'm not entirely certain, that you can get way with not quoting the tag.

**Review by fabiomadge — APPROVED** [review]

**Pull request merged**